Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1511,7 +1511,7 @@ LayoutUnit RenderFlexibleBox::marginBoxAscentForFlexItem(const RenderBox& flexIt
     }
 
     if (isHorizontalFlow ? flexItem.isScrollContainerY() : flexItem.isScrollContainerX())
-        return std::clamp(*ascent, 0_lu, crossAxisExtentForFlexItem(flexItem)) + flowAwareMarginBeforeForFlexItem(flexItem);
+        return std::clamp(*ascent, 0_lu, std::max(0_lu, crossAxisExtentForFlexItem(flexItem))) + flowAwareMarginBeforeForFlexItem(flexItem);
```

Comment:
I would instead replace clamp with min/max.

Reply:
I had that earlier but @rniwa suggested this is simpler.

Reply:
it may look simpler but I think the clamp usage here is incorrect as we can't guarantee a proper range.  We normally (and historically) use min/max for such cases as opposed to massaging the range (also this change can be seen as more error prone as now we've got to change the range value at 2 places if e.g. for whatever reason range is supposed to start at 1 instead of 0)

Reply:
Thanks Alan for the explanation. I have updated the PR.